lib/neo4j/railtie.rb

```diff
@@ -74,8 +75,10 @@ def final_config!(neo4j_config)
     end
 
     def support_deprecated_session_configs!(neo4j_config)
-      ActiveSupport::Deprecation.warn('neo4j.config.sessions is deprecated, please use neo4j.config.session (not an array)') if neo4j_config.sessions.present?
-      neo4j_config.session ||= (neo4j_config.sessions && neo4j_config.sessions[0]) || {}
+      if neo4j_config.sessions.present?
```

Comment:
But that's exactly what this code is doing. I am assuming you want to keep old configurations working. If someone has configured `sessions` we will take the first one. Newest master is now in.
My refactoring of your code is equivalent expect that I removed that hidden initialization side effect.